CentralComputing/Pod.cpp

```diff
@@ -1,104 +1,35 @@
-		TRANSITION_MAP_ENTRY(EVENT_IGNORED)			/* Flight coast */
-		TRANSITION_MAP_ENTRY(ST_SAFE_MODE)			/* Flight brake */
-	END_TRANSITION_MAP(NULL)
-}
-
-void Pod::move_flight_acceleration() {
-	BEGIN_TRANSITION_MAP							/* Current state */
-		TRANSITION_MAP_ENTRY(EVENT_IGNORED)			/* Safe mode */
-		TRANSITION_MAP_ENTRY(ST_FLIGHT_ACCEL)		/* Functional test */
-		TRANSITION_MAP_ENTRY(ST_FLIGHT_ACCEL)		/* Flight accel */
-		TRANSITION_MAP_ENTRY(EVENT_IGNORED)			/* Flight coast */
-		TRANSITION_MAP_ENTRY(EVENT_IGNORED)			/* Flight brake */
-	END_TRANSITION_MAP(NULL)
-}
-
-// it is important all states should move to braking when this function is called, this is for emergencies
-void Pod::emergency_brake() {
-	BEGIN_TRANSITION_MAP							/* Current state */
-		TRANSITION_MAP_ENTRY(ST_FLIGHT_BRAKE)		/* SAFE MODE */
-		TRANSITION_MAP_ENTRY(ST_FLIGHT_BRAKE)		/* FUNCTIONAL TEST */
-		TRANSITION_MAP_ENTRY(ST_FLIGHT_BRAKE)		/* FLIGHT ACCEL */
-		TRANSITION_MAP_ENTRY(ST_FLIGHT_BRAKE)		/* FLIGHT COAST */
-		TRANSITION_MAP_ENTRY(ST_FLIGHT_BRAKE)		/* FLIGHT BRAKE */
-	END_TRANSITION_MAP(NULL)
-}
+using namespace std;
 
+Sensor_Package * sensors;
 /**
- * Software controlled events
- **/
-void Pod::coast() {
-	BEGIN_TRANSITION_MAP							/* Current state */
-		TRANSITION_MAP_ENTRY(EVENT_IGNORED)		/* SAFE MODE */
-		TRANSITION_MAP_ENTRY(EVENT_IGNORED)		/* FUNCTIONAL TEST */
-		TRANSITION_MAP_ENTRY(ST_FLIGHT_COAST)	/* FLIGHT ACCEL */
-		TRANSITION_MAP_ENTRY(ST_FLIGHT_COAST)	/* FLIGHT COAST */
-		TRANSITION_MAP_ENTRY(EVENT_IGNORED)		/* FLIGHT BRAKE */
-	END_TRANSITION_MAP(NULL)
-}
-
-void Pod::brake() {
-	BEGIN_TRANSITION_MAP						/* Current state */
-		TRANSITION_MAP_ENTRY(EVENT_IGNORED)		/* SAFE MODE */
-		TRANSITION_MAP_ENTRY(EVENT_IGNORED)		/* FUNCTIONAL TEST */
-		TRANSITION_MAP_ENTRY(EVENT_IGNORED)		/* FLIGHT ACCEL */
-		TRANSITION_MAP_ENTRY(ST_FLIGHT_BRAKE)	/* FLIGHT COAST */
-		TRANSITION_MAP_ENTRY(ST_FLIGHT_BRAKE)	/* FLIGHT BRAKE */
-	END_TRANSITION_MAP(NULL)
+* Gets new sensor values from the XMEGA
+*/
+void sensor_loop() {
+	//TODO construct XMEGA transfer object
+	//sensors->update();
 }
 
+int pod(int argc, char** argv) {
+	
+	vector<Sensor_Configuration> configs = vector<Sensor_Configuration>();
 
-// State Machine State functions
-void Pod::ST_Safe_Mode() {
-	std::cout << "Entering: Safe Mode" << std::endl;
-	//std::cout << GetCurrentState() << std::endl;
-	// TODO implement here
-}
-
-void Pod::ST_Functional_Test() {
-	std::cout << "Entering: Functional Test" << std::endl;
-	// TODO implement here
-}
-
-void Pod::ST_Flight_Accel() {
-	std::cout << "Entering: Flight Accel" << std::endl;
-
-	// TODO implement here
-}
+	//sets up thermocouple configuration
+	Sensor_Configuration thermo;
+	thermo.type = THERMOCOUPLE;
+	thermo.simulation = 0;
+	thermo.count = 4;
 
-void Pod::ST_Flight_Coast() {
-	std::cout << "Entering: Flight Coast" << std::endl;
+	//TODO set up other configurations
 	
-	// TODO implement here
-}
+	sensors = new Sensor_Package(configs);
```

Comment:
Does it make sense to just have the `Sensor_Package` variable defined in the thread where it will be used? Does it need to be a global variable. Could it just be passed into the thread as an argument? I am generally against global variables when not necessary